addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java

```diff
@@ -123,7 +123,8 @@ public void init() {
     @Override
     public void scan() {
         if ((AlertThreshold.HIGH.equals(getAlertThreshold()) && !getBaseMsg().isInScope())
-                || !getBaseMsg().getResponseHeader().isHtml()) {
+                || !getBaseMsg().getResponseHeader().isHtml()
+                || isPage404(getBaseMsg())) {
```

Comment:
So your assumption is a 404 response can never be subject to CSRF?

Reply:
Yes. Isn't is true for majority cases?

Reply:
Largely, I didn't mean it was a problem I just wanted to be on the same page.

Reply:
Yes we are on the same page.